modules/consensus/consensusset.go

```diff
@@ -176,7 +177,10 @@ func (cs *ConsensusSet) ChildTarget(id types.BlockID) (target types.Target, exis
 func (cs *ConsensusSet) Close() error {
 	cs.mu.Lock()
 	defer cs.mu.Unlock()
-	return cs.db.Close()
+	return build.JoinErrors([]error{
+		build.AddContext("log.Close failed:", cs.log.Close()),
+		build.AddContext("db.Close failed:", cs.db.Close()),
+	}, "; ")
```

Comment:
A disadvantage of the `build.JoinErrors` pattern here is that you can only call functions that only return an error, and you cannot intersperse other function calls (such as locking / unlocking).

Then again, maybe it is good style to enforce that the components of the module be able to close easily like this, as it makes it easier to reason about.